### ISSUE TITLE
Depend on es6-collections+es6-promise rather than es6-shim

### ISSUE THREAD

**Issue by alexeagle** [angular/angular#7124]
We now type-check against the es6-collections typings rather than es6-shim.
The only exception is we use a small number of methods indicated in https://github.com/angular/angular/blob/master/modules/angular2/manual_typings/globals.d.ts
which we ought to remove.

To let users build a smaller application, we should not require polyfills for all of es6. They can just use a polyfill for es6-promise and es6-collections.

Reminded of this by Igor's comment https://github.com/angular/angular/issues/6659#issuecomment-185120027

**Comment by SIGAN** [comment]
But there is a the problem with es6-collections & es6-promise. 

If you use them, users can't use es6-shim, because it will end up with double declaration of Promise & Collections.

TypeScript then won't compile!

How to get then newer array or string methods?

**Comment by alexeagle** [comment]
Users don't have to use es6-collections or es6-promise (in fact if you
follow the quickstart right now, you don't)

**Comment by IgorMinar** [comment]
I believe this is now obsolete. If I missed something please open a new issue.

**Comment by alexeagle** [comment]
For the record, I still believe we are doing the wrong thing, and forcing users to install more polyfills than should be required.
